subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -712,24 +724,27 @@ static void response_cb(struct http_response *rsp,
 
 				hb_context.response_data = rsp_tmp;
 			}
-
-			strncat(hb_context.response_data, body_data,
-				rsp->data_len);
+			strncpy(hb_context.response_data + hb_context.dl.downloaded_size,
+				body_data, body_len);
+			hb_context.dl.downloaded_size += body_len;
 		}
 
 		if (final_data == HTTP_DATA_FINAL) {
-			if (hb_context.dl.http_content_size != body_len) {
-				LOG_ERR("HTTP response len mismatch");
+			if (hb_context.dl.http_content_size != hb_context.dl.downloaded_size) {
+				LOG_ERR("HTTP response len mismatch, expected %d, got %d",
+					hb_context.dl.http_content_size,
+					hb_context.dl.downloaded_size);
 				hb_context.code_status = HAWKBIT_METADATA_ERROR;
```

Comment:
@Navin-Sankar I wonder for this kind of error should it just `cleanup_connection()` and `break` just like [here](https://github.com/zephyrproject-rtos/zephyr/blob/2171df5bfd55bb7897a9eb77e507d3149bafc60d/subsys/mgmt/hawkbit/hawkbit.c#L721)?